File: lib/ts/src/internal/UnknownError.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { Message } from '../Message.js';
import { TelepactError } from '../TelepactError.js';

export function buildUnknownErrorMessage(error: TelepactError, headers: Record<string, any> = {}): Message {
    return new Message(headers, { ErrorUnknown_: { caseId: error.caseId } });
}

File: lib/py/telepact/internal/UnknownError.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from ..Message import Message
from ..TelepactError import TelepactError

def build_unknown_error_message(
    error: TelepactError,
    headers: dict[str, object] | None = None,
) -> Message:
    return Message(headers or {}, {"ErrorUnknown_": {"caseId": error.case_id}})
